teps/0048-task-results-without-results.md

```diff
@@ -0,0 +1,425 @@
+  [bug report](https://github.com/tektoncd/pipeline/issues/3497) open to declare
+  such a task as failure. This reason would not hold true after issue
+  [#3497]((https://github.com/tektoncd/pipeline/issues/3497)) is fixed.
+
+As we are expanding task results consumption within `finally` tasks and adding support for
+[continueAfterSkip](https://github.com/tektoncd/community/pull/258),
+it becomes crucial to define how a `pipeline` can handle such missing results. In this proposal, we are evaluating
+different possible solutions and determine which one fits best to address this issue.
+
+## Motivation
+
+Missing task results do not have to be fatal. Give pipeline authors an option to build pipeline that can continue
+executing when a task result is missing.
+
+### Goals
+
+Identify a set of use cases for `finally` and `continueAfterSkip` without task results and design a common solution
+to address those use cases.
+
+### Non-Goals
+
+This proposal is not offering a solution without any use case.
+
+## Proposal
+
+Enable `pipeline` author to specify a default value for a `task result` in case of a missing result. Pipeline author
+can specify default for the `param` referring to task result if that `param` is declared optional in the `task`.
+
+Generally, task author designs a task with set of inputs in the form of `params` while producing some results in the
+form of `results`. When a `task` is incorporated in a `pipeline`, `pipeline` author has control over these `params`
+and can design it such that the `params` are mapped to `pipeline` params or initialized with the `results` of other
+tasks in a `pipeline`. At the Task level, parameters can be declared as optional. The Task author may provide a default
+value for optional parameters. The Pipeline author must populate every parameter that is not declared as optional in the Task.
+
+This proposal introduces a new field called `results` at the `pipelineTask` level. `results` can have a list of result
+variables, and their default values.
+
+In the following example, `pipelineTask` `add` has specified `results` with the default value. Now, `x` is referring
+to a task result of `add`. If `add` fails for some reason, finally task `multiply` continues execution with default
+`sum` which is `0`:
+
+```yaml
+kind: Pipeline
+spec:
+  tasks:
+    - name: add
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          default: 0
+  finally:
+    - name: multiply
+      taskRef:
+        name: multiply
+      params:
+        - name: x
+          value: $(task.add.results.sum)
+```
+
+### User Stories
+
+#### Story 1
+
+For a cleanup use case, `finally` task receives the project name, or the configuration name as a `task result` from
+the `acquire` task. If `acquire` task fails to acquire any resources, `cleanup` is skipped.
```

Comment:
i think there's some detail missing here - it seems like if the resources weren't acquired, you don't want to do the cleanup?